### PR TITLE
Explain RB tree rotations a bit more.

### PR DESCRIPTION
**Pull request by justhsu**
Say a bit more about the RB tree rotations and why they are correct. @clarksmr do you think this is clearer?

### CODE REVIEW

File: src/chapters/ds/rb.md [hunk 1]
```diff
@@ -185,7 +185,10 @@ black.
 
 The next figure shows the four possible cases that can arise. In it, `a`-`d` are
 possibly empty subtrees, and `x`-`z` are values stored at a node. The nodes
-colors are indicated with `R` and `B`.
+colors are indicated with `R` and `B`. We've marked the current "new" node with
```
Comment (clarksmr) [inline src/chapters/ds/rb.md:188]: Hm, to me the idea of `current "new" node` is confusing. It is no longer necessarily a new node.

File: src/chapters/ds/rb.md [hunk 2]
```diff
@@ -230,10 +233,21 @@ replacing any of the above four cases with:
...
+hence that the final tree is the same in all four cases.
+```
+
+```{tip}
+It’s also worth thinking about why these operations preserves the local and
+global invariants. We have to use a careful combination of facts about the
+various subtrees. For instance, in the first case:
+
+- `x` is the current “new” node, and the subtree starting from `x` satisfies all RB tree invariants.
+- the subtree `c` satisfies all RB tree invariants.
```
Comment (clarksmr) [inline src/chapters/ds/rb.md:248]: What about `d`?

File: src/chapters/ds/rb.md [hunk 3]
```diff
@@ -171,8 +171,15 @@ Here's how it works.
 
 We always color the new node red to ensure that the Global Invariant is
 preserved. However, this may destroy the Local Invariant by producing two
-adjacent red nodes. In order to restore the invariant, we consider not only the
-new red node and its red parent, but also its (black) grandparent.
+adjacent red nodes. If the parent of the new node is the root, we can simply
+change the color of the root to be black. This restores the local invariant
```
Comment (clarksmr) [inline src/chapters/ds/rb.md:175]: In Okasaki's conventions, which we follow here, the root must already be black. So we wouldn't change its color.

Comment (justhsu) [reply src/chapters/ds/rb.md:175]: OK, I think I was confused about this. Should the root being black be part of the RI, then?

File: src/chapters/ds/rb.md (hunk 1 above)
Comment (justhsu) [reply src/chapters/ds/rb.md]: Yeah, I don't have a good name. I'm now calling it the "root of the subtree that we've fixed up so far", the "marked node" for short.

**Comment by clarksmr** [comment]
@justhsu I'm optimistically assuming you're okay with me merging this now, after my further edits. If you want to take a look and make further suggestions that's fine too!

**Pull request merged**